in_toto/rulelib.py

```diff
@@ -167,10 +155,133 @@ def unpack_rule(rule):
           "Got: \n\t{}".format(rule))
 
     return {
-      "type": rule_type,
+      "rule_type": rule_type,
       "pattern": pattern,
       "source_prefix": source_prefix,
       "dest_prefix": dest_prefix,
       "dest_type" : dest_type,
       "dest_name": dest_name
     }
+
+
+def pack_rule(rule_type, pattern, source_prefix=None, dest_type=None,
+      dest_prefix=None, dest_name=None):
+  """
+  <Purpose>
+    Create an artifact rule in the passed arguments and return it as list
+    as it is stored in a step's or inspection's expected_material or
+    expected_product field.
+
+  <Arguments>
+    rule_type:
+            One of "MATCH", "CREATE", "DELETE", MODIFY, ALLOW, DISALLOW
+            (case insensitive).
+
+    pattern:
+            A glob pattern to match artifact paths.
+
+    source_prefix: (only for "MATCH" rules)
+            A prefix for 'pattern' to match artifacts reported by the link
+            corresponding to the step that contains the rule.
+
+    dest_type: (only for "MATCH" rules)
+            One of "MATERIALS" or "PRODUCTS" (case insensitive) to specify
+            if materials or products of the link corresponding to the step
+            identified by 'dest_name' should be matched.
+
+    dest_prefix: (only for "MATCH" rules)
+            A prefix for 'pattern' to match artifacts reported by the link
+            corresponding to the step identified by 'dest_name'.
+
+    dest_name: (only for "MATCH" rules)
+            The name of the step whose corresponding link is used to match
+            artifacts.
+
+
+  <Exceptions>
+    securesystemslib.exceptions.FormatError
+            if any of the arguments is malformed.
+
+
+  <Returns>
+    One of the following rule formats in as a list
+      MATCH <pattern> [IN <source-path-prefix>] WITH (MATERIALS|PRODUCTS)
+          [IN <destination-path-prefix>] FROM <step>,
+      CREATE <pattern>,
+      DELETE <pattern>,
+      MODIFY <pattern>,
+      ALLOW <pattern>,
+      DISALLOW <pattern>
+
+  """
+  in_toto.formats.ANY_STRING_SCHEMA.check_match(rule_type)
```

Comment:
I don't know if it's overkill, but we could probably have an `MATCH_RULE_SCHEMA` using an `anyof` of `ALL_RULES` and simplify the if below. Specially considering we are going heavy on the scheme checkers.

Reply:
Definitely a good idea to use schema. But maybe worth an extra refactor ticket?